source/main/cardano/CardanoSelfnodeLauncher.ts

```diff
@@ -86,11 +106,30 @@ export async function CardanoSelfnodeLauncher(
           SHELLEY_TEST_DATA,
           TOKEN_METADATA_SERVER,
         },
-        detached: true,
+        detached: environment.isDev, // XXX: detaching breaks Windows launching the local-cluster.exe + cardano-launcher
+        // will not allow you to start another Daedalus when previous local-cluster.exe is running, especially awkward
+        // on macOS without stdout/stderr logging; I’m not sure if this is ever needed in `nix-shell`, but leaving as-is
+        // for now – @michalrus
+
         // allows Daedalus to exit independently of selfnode (1/3)
-        stdio: 'ignore', // allows Daedalus to exit independently of selfnode (2/3)
+        stdio: environment.isDev ? 'ignore' : 'pipe', // 'ignore' allows Daedalus to exit independently of selfnode, 'pipe' kills it on exit (2/3)
       });
-      nodeProcess.unref(); // allows Daedalus to exit independently of selfnode (3/3)
+      if (environment.isDev) {
+        nodeProcess.unref(); // allows Daedalus to exit independently of selfnode (3/3)
+      }
+      if (
+        !environment.isDev &&
+        nodeProcess.stdout &&
+        nodeProcess.stderr &&
+        walletLogFile
+      ) {
+        nodeProcess.stdout.on('data', (data) => {
+          walletLogFile.write(data);
+        });
+        nodeProcess.stderr.on('data', (data) => {
+          walletLogFile.write(data);
```

Comment:
I’m not sure why I can’t just do `nodeProcess.stderr.on('data', walletLogFile.write)` – perhaps because of the 2nd optional parameter of `write`? Inconsistent syntax, tho…